lib/cc/yaml/nodes/engine.rb

```diff
@@ -4,7 +4,28 @@ module Nodes
       class Engine < Mapping
         map :enabled, to: Scalar[:bool], required: true
         map :checks, to: Checks
-        map :config, to: EngineConfig
+        map :config, to: Mapping
+
+        attr_reader :config
+
+        def visit_key_value(visitor, key, value)
+          if key == "config"
+            raw_value = value.to_ruby
+
+            # backwards compatability for phpcodesniffer
+            if raw_value.is_a?(String)
+              @config = { "file" => raw_value }
+            else
+              @config = raw_value
+            end
+          else
+            super
+          end
+        end
+
+        def reload
+          self.merge!("config" => @config)
```

Comment:
I've tried modyfing `#visit_key_value` method before you suggested and couldn't quite make it work. The `reload` is used by Codeclimate CLI, where at some point, a config for each engine is retrieved. Because `config` can be accessed only via attr_reader, it would require modifications in the CLI code. `reload` simply converts the Engine Node into hash, which would be done anyway, but config values would be stripped. `reload` prevents that. @pbrisbin